pkg/providers/amifamily/suite_test.go

```diff
@@ -303,6 +303,138 @@ var _ = Describe("AMIProvider", func() {
 			}))
 		})
 	})
+	Context("AMI List requirements", func() {
+		BeforeEach(func() {
+			nodeClass.Spec.AMISelectorTerms = []v1.AMISelectorTerm{
+				{
+					Tags: map[string]string{"*": "*"},
+				},
+			}
+		})
+		It("should priortize the older non-deprecated ami without deprecation time", func() {
+			// Here we have two AMIs one which is deprecated and newer and one which is older and non-deprecated without a deprecation time
+			// List operation will priortize the non-deprecated AMI without the deprecation time
+			awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
+				Images: []*ec2.Image{
+					{
+						Name:            aws.String(amd64AMI),
+						ImageId:         aws.String("ami-5678"),
+						CreationDate:    aws.String("2021-08-31T00:12:42.000Z"),
+						DeprecationTime: aws.String(time.Now().Add(-1 * time.Minute).Format(time.RFC3339)),
+						Architecture:    aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String(amd64AMI)},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+					{
+						Name:         aws.String(amd64AMI),
+						ImageId:      aws.String("ami-1234"),
+						CreationDate: aws.String("2020-08-31T00:08:42.000Z"),
+						Architecture: aws.String("x86_64"),
+						Tags: []*ec2.Tag{
+							{Key: aws.String("Name"), Value: aws.String(amd64AMI)},
+							{Key: aws.String("foo"), Value: aws.String("bar")},
+						},
+					},
+				},
+			})
+			amis, err := awsEnv.AMIProvider.List(ctx, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(amis).To(HaveLen(1))
+			Expect(amis).To(ConsistOf(amifamily.AMI{
+				Name:         amd64AMI,
+				AmiID:        "ami-1234",
+				CreationDate: "2020-08-31T00:08:42.000Z",
+				Requirements: scheduling.NewRequirements(
+					scheduling.NewRequirement(corev1.LabelArchStable, corev1.NodeSelectorOpIn, karpv1.ArchitectureAmd64),
+				),
+			}))
+		})
+		It("should priortize the non-deprecated ami with deprecation time when both have same creation time", func() {
+			// Here we have two AMIs one which is deprecated and one which is non-deprecated both with the same creation time
+			// List operation will priortize the non-deprecated AMI
+			awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
```

Comment:
I think this will fail if depending on the naming the of the AMI, no?

Reply:
Added an additional test